### PR TITLE
Search for new aeternity config locations

### PR DESCRIPTION
**Pull request by zp-sd**
Corresponding infrastructure PR - https://github.com/aeternity/infrastructure/pull/229.

System discovery in the logs:
```
11:55:51.481 [info] Searching for config file aeternity.{json,yaml} in directory /Users/zp/.aeternity/aeternity
11:55:51.481 [info] Searching for config file aeternity.{json,yaml} in directory /Users/zp/src/aeternity/aeternity
11:55:51.481 [info] Searching for config file epoch.{json,yaml} in directory /Users/zp/.epoch/epoch
11:55:51.481 [info] Searching for config file epoch.{json,yaml} in directory /Users/zp/src/aeternity/aeternity
11:55:51.481 [info] No config file specified; using default settings
```

All ways of passing config tested locally.

System tests:
```
===> Running Common Test suites...
%%% aest_channels_SUITE ==> test_simple_same_node_channel: OK
%%% aest_channels_SUITE ==> test_simple_different_nodes_channel: OK
%%% aest_channels_SUITE ==> on_chain_channel: OK
%%% aest_commands_SUITE ==> epoch_commands: OK
%%% aest_compatibility_SUITE ==> test_mining_algorithms_compatibility: OK
%%% aest_genesis_SUITE ==> test_node_starts_with_30k_accounts: OK
%%% aest_mempool_SUITE ==> test_mempool_ttl_cleanup: OK
%%% aest_mempool_SUITE ==> test_mempool_bad_nonce_cleanup: OK
%%% aest_names_SUITE ==> test_name_registration: OK
%%% aest_names_SUITE ==> test_name_transfer: OK
%%% aest_oracles_SUITE ==> test_simple_same_node_query: OK
%%% aest_oracles_SUITE ==> test_simple_two_nodes_query: OK
%%% aest_oracles_SUITE ==> test_oracle_ttl_extension: OK
%%% aest_oracles_SUITE ==> test_pipelined_same_node_query: OK
%%% aest_oracles_SUITE ==> test_pipelined_two_nodes_query: OK
%%% aest_peers_SUITE ==> test_peer_discovery: OK
%%% aest_peers_SUITE ==> test_inbound_limitation: OK
%%% aest_perf_SUITE ==> long_chain.long_chain_mine_and_export: OK
%%% aest_perf_SUITE ==> long_chain.long_chain_tests.mining_governed_rate: OK
%%% aest_perf_SUITE ==> long_chain.long_chain_tests.startup_speed: OK
%%% aest_perf_SUITE ==> long_chain.long_chain_tests.startup_speed_mining: OK
%%% aest_perf_SUITE ==> long_chain.long_chain_tests.sync_speed: OK
%%% aest_perf_SUITE ==> long_chain.long_chain_tests.stay_in_sync: OK
%%% aest_sync_SUITE ==> new_node_joins_network: OK
%%% aest_sync_SUITE ==> docker_keeps_data: OK
%%% aest_sync_SUITE ==> stop_and_continue_sync: OK
%%% aest_sync_SUITE ==> tx_pool_sync: OK
%%% aest_sync_SUITE ==> net_split_recovery: OK
%%% aest_sync_SUITE ==> net_split_mining_power: OK
%%% aest_sync_SUITE ==> abrupt_stop_new_node: SKIPPED
%%% aest_sync_SUITE ==> {tc_user_skip,database_restart_needs_fix}
%%% aest_sync_SUITE ==> abrupt_stop_mining_node: SKIPPED
%%% aest_sync_SUITE ==> {tc_user_skip,database_restart_needs_fix}
EXPERIMENTAL: Writing retry specification at /Users/zp/src/aeternity/aeternity/system_test/logs/retry.spec
              call rebar3 ct with '--retry' to re-run failing cases.
Skipped 2 (2, 0) tests. Passed 29 tests.
```

### CODE REVIEW

File: apps/aeutils/src/aeu_env.erl [hunk 1]
```diff
@@ -286,18 +297,30 @@ config_file() ->
...
+    end.
 
 search_default_config() ->
-    Dirs = [filename:join([os:getenv("HOME"), ".epoch", nodename()]),
+    Dirs = [filename:join([os:getenv("HOME"), ".aeternity", "aeternity"]),
             setup:home()],
+    search_config(Dirs, "aeternity.{json,yaml}").
+
+search_deprecated_config() ->
+    Dirs = [filename:join([os:getenv("HOME"), ".epoch", "epoch"]),
```
Comment (dincho) [inline apps/aeutils/src/aeu_env.erl]: possible BC break, but IIRC we agreed it's OK - very low probability

File: docs/release-notes/next/PT-163191737-config-discovery-renaming.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+* Changes user config discovery paths, i.e. the node is looking for the user config in:
+  - AETERNITY_CONFIG environment variable instead of EPOCH_CONFIG,
+  - ~/.aeternity/aeternity/aeternity.yaml file instead of ~/.epoch/epoch/epoch.yaml,
+  - ${AETERNITY_TOP}/aeternity.yaml file instead of ${AETERNITY_TOP}/epoch.yaml.
+  Backwards compatibility is kept for now, so user config defined in old locations will be working.
```
Comment (dincho) [inline docs/release-notes/next/PT-163191737-config-discovery-renaming.md]: until next major release

File: apps/aeutils/src/aeu_env.erl (hunk 1 above)
Comment (zp-sd) [reply apps/aeutils/src/aeu_env.erl]: actually not possible until someone manipulated with `vm.args` by hand...

File: docs/release-notes/next/PT-163191737-config-discovery-renaming.md (hunk 2 above)
Comment (zp-sd) [reply docs/release-notes/next/PT-163191737-config-discovery-renaming.md]: @lucafavatella Can this be better worded either when preparing the release or in the separate PR? I'd rather not loose approvals by this change...

**Pull request merged**